content/en/blog/2022/k8sattributes.md

```diff
@@ -0,0 +1,256 @@
+[OpenTelemetry collector documentation](/docs/collector/).
+
+## How k8s attributes are attached
+
+At a high level, k8s attributes are attached to traces as
+[Resources](/docs/concepts/glossary/#resource). This is for two reasons:
+
+1.  They fit the definition of what a Resource is - an entity for which
+    telemetry is recorded
+2.  Centralizes this metadata, which is relevant for any span generated
+
+Let's dive in and see how to do it!
+
+## Using k8sattributes processor
+
+The `k8sattributesprocessor` is an OpenTelemetry collector processor that
+attaches Kubernetes attributes to spans, logs and metrics.
+
+This processor automatically discovers pod attributes and attaches those
+attributes to resource asociated with the spans generated by that pod. If the
+pod belongs to a `Deployment` or a `ReplicaSet`, it will also discover it's
+attributes.
+
+Some attributes we can attach to the resource are:
+
+- node name `k8s.node.name`.
+- pod name `k8s.pod.name`.
+- pod UID `k8s.pod.uid`.
+- namespace `k8s.namespace.name`.
+- deployment name, if the pod was created by a deployment.
+
+Those attributes use the OpenTelemetry semantic convention. You can see a full
+list
+[here](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/resource/semantic_conventions/k8s.md).
+
+The processor internally maintains a list of pods and an associated attribute,
+usually the IP address of the pod, and uses this attribute to know which pod
+generates a certain span.
+
+![k8sattributes processor data flow](/img/blog-k8sattributes/k8sprocessor.png)
+
+In the figure above you can see how the data flows: The table of pods is fetched
+using Kubernetes API, while the pod IP is extracted from the connection context
+between the pod and the collector.
+
+The `k8sattributesprocessor` can work in different modes, and it depends on how
+the collector is configured. Let's explore one common scenario, when the
+collector is deployed as daemonset.
+
+### DaemonSet mode
+
+Let’s take a look at how we can configure the collector in daemonset mode, known
+as an agent mode in the k8sattributes documentation.
+
+When we deploy the collector as the DaemonSet we have one collector pod per
+node. We need to configure the collector service account to have permissions to
+fetch all pods information. In order to do that, we will create a `ClusterRole`
+with the necessary permissions.
+
+Here are the minimum permissions required for make k8sattributes processor
+works:
+
+```yaml
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: otel-collector
+rules:
+  - apiGroups: [""]
+    resources: ["pods", "namespaces"]
+    verbs: ["get", "watch", "list"]
+```
+
+Now, we are going to deploy the collector in daemonset mode. It is recommended
+to set a filter to only fetch the pods that belong to the node in which the
+collector is deployed. This is because if we are on a large cluster, we don’t
+want to maintain a huge list of pods.
```

Comment:
```suggestion
Next, deploy the collector in daemonset mode. It is recommended
to set a filter to only fetch the pods that belong to the node in which the
collector is deployed. This is because if you have a large cluster, you don’t
want to maintain a huge list of pods.
```